### PR TITLE
Feat(ui) Added data privacy button in issues page

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Description:** 

We wanted to provide the user the ability of adding a new data privacy rule from the issue details page, with the event ID pre-filled.

ps: This PR moves data privacy rules components that were before views/settings - only to the folder app/components/dataPrivacyRules

image

image

TODO:
Replace some of the dataPrivacyRules components with  the new ones from the folder app/components/dataPrivacyRules

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dataPrivacyRules/sourceSuggestionExamples.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import Modal from 'react-bootstrap/lib/Modal';
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/sourceSuggestionExamples.tsx:1]: moved code

File: src/sentry/static/sentry/app/components/group/suggestedOwners/styles.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import space from 'app/styles/space';
 
 const Heading = styled('h6')`
   margin: 0 !important;
-  font-weight: 600;
+  font-weight: 400;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/suggestedOwners/styles.tsx:7]:
```suggestion
  font-weight: 600;
```

File: src/sentry/static/sentry/app/components/dataPrivacyRules/selectField.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import styled from '@emotion/styled';
+// eslint import checks can't find types in the flow code.
+// eslint-disable-next-line import/named
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/selectField.tsx:4]: Is this a warning or error? For warnings I think we would prefer to leave them.

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dataPrivacyRules/selectField.tsx:4]: it's an error: /Users/priscila/Projects/sentry/src/sentry/static/sentry/app/components/dataPrivacyRules/selectField.tsx

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dataPrivacyRules/selectField.tsx:4]: its an error 
image

File: src/sentry/static/sentry/app/components/dataPrivacyRules/dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+type State = {
+  rule: Rule;
+  isFormValid: boolean;
+  errors: Errors;
+};
+
+class Dialog extends React.Component<Props, State> {
+  state: State = {
+    rule: {
+      id: defined(this.props.rule?.id) ? this.props.rule?.id! : -1,
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/dialog.tsx:48]: What does an id of -1 signify? Should you use an absent value like undefined instead?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    switch (status) {
+      case EventIdStatus.INVALID:
+        return t('This event ID is invalid.');
+      case EventIdStatus.ERROR:
+        return t(
+          'An error occurred while fetching the suggestions based on this Event ID.'
+        );
+      case EventIdStatus.NOT_FOUND:
+        return t('The chosen event ID was not found in projects you have access to.');
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx:98]:
```suggestion
        return t('The chosen Event ID was not found in projects you have access to.');
```

Event ID be capitalized consistently if we're going to title case it.

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      status: undefined,
+    });
+  };
+
+  getErrorMessage = (): string | undefined => {
+    const {status} = this.state;
+
+    switch (status) {
+      case EventIdStatus.INVALID:
+        return t('This event ID is invalid.');
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx:92]:
```suggestion
        return t('This Event ID is invalid.');
```

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+  render() {
+    const {disabled} = this.props;
+    const {value, status} = this.state;
+
+    return (
+      <FormField
+        label={t('Event ID (Optional)')}
+        tooltipInfo={t(
+          'Providing an event ID will automatically provide you a list of suggested sources'
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx:112]:
```suggestion
          'Providing an Event ID will automatically provide you a list of suggested sources'
```

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        <CloseIcon onClick={onClickIconClose}>
+          <Tooltip title={t('Clear Event ID')}>
+            <IconClose color="red" />
+          </Tooltip>
+        </CloseIcon>
+      );
+    case EventIdStatus.LOADING:
+      return <ControlState isSaving />;
+    case EventIdStatus.LOADED:
+      return <IconCheckmark color="green" />;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx:31]: Should we use a color from the theme instead?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+const EventIdFieldStatusIcon = ({status, onClickIconClose}: Props) => {
+  switch (status) {
+    case EventIdStatus.ERROR:
+    case EventIdStatus.INVALID:
+    case EventIdStatus.NOT_FOUND:
+      return (
+        <CloseIcon onClick={onClickIconClose}>
+          <Tooltip title={t('Clear Event ID')}>
+            <IconClose color="red" />
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx:24]: Should we use a color from the theme instead?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/handleError.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+// TODO(ts): define the correct error type
+function handleError(error: any): Error {
+  const errorMessage = error.responseJSON?.relayPiiConfig[0];
+
+  if (!errorMessage) {
+    return {
+      type: RequestError.Unknown,
+      message: t('Unknown error occurred while saving data privacy rules'),
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/handleError.tsx:17]:
```suggestion
      message: t('An unknown error occurred while saving data privacy rules'),
```

File: src/sentry/static/sentry/app/components/dataPrivacyRules/handleError.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+          type: RequestError.RegexParse,
+          message: t('Invalid regex: %s', regex),
+        };
+      }
+    }
+  }
+
+  return {
+    type: RequestError.Unknown,
+    message: t('Unknown error occurred while saving data privacy rules'),
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/handleError.tsx:47]:
```suggestion
    message: t('An unknown error occurred while saving data privacy rules'),
```

File: src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx [hunk 12]
```diff
@@ -0,0 +1,463 @@
...
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 100%;
+  height: 40px;
+  input {
+    height: 40px;
+  }
+
+  z-index: 1002;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx:412]: Are we able to use theme.zIndex here?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx [hunk 13]
```diff
@@ -0,0 +1,463 @@
...
+  padding-left: 0;
+  list-style: none;
+  margin-bottom: 0;
+  box-shadow: 0 2px 0 rgba(37, 11, 54, 0.04);
+  border: 1px solid ${p => p.theme.borderDark};
+  border-radius: 0 0 ${space(0.5)} ${space(0.5)};
+  background: ${p => p.theme.white};
+  top: 35px;
+  right: 0;
+  z-index: 1002;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx:430]: Are we able to use theme.zIndex here?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx [hunk 14]
```diff
@@ -0,0 +1,463 @@
...
+  color: ${p => p.theme.gray2};
+`;
+
+const SuggestionsOverlay = styled('div')`
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  z-index: 1001;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/sourceField.tsx:462]: Are we able to use theme.zIndex here?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/sourceSuggestionExamples.tsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+export default SourceSuggestionExamples;
+
+const StyledModal = styled(Modal)`
+  .modal-dialog {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%) !important;
+    margin: 0;
+    z-index: 1003;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/sourceSuggestionExamples.tsx:68]: Are we able to use theme.zIndex here?

File: src/sentry/static/sentry/app/components/dataPrivacyRules/utils.tsx [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+  switch (type) {
+    case MethodType.MASK:
+      return {
+        label: t('Mask'),
+        description: t('Replace with ****'),
+      };
+    case MethodType.HASH:
+      return {
+        label: t('Hash'),
+        description: t('Replace with DEADBEEF'),
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dataPrivacyRules/utils.tsx:48]: Does this actually use a literal deadbeef? Or does it replace with the hashed contents?

**Comment by dcramer** [comment]
Let's not start referring to this as data privacy, and have a product review this larger change (or set of changes). We can't rename functions like this as that has meaning, and also would require updating numerous docs. Additionally, data scrubbing is not just for data privacy, its also for security.

Lastly, why do we think this needs to be on the issue details page? Why is this so much higher value than other actions? It seems fairly subjective and I'd rather not have the sidebar loaded up with every setting you can possibly change. The reason the ownership rule lives right there is because the Assignee selector is right there. The same cannot be said about this suggested action.

File: src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx (hunk 8 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdFieldStatusIcon.tsx:31]: it is from the theme 😉 

image

**Comment by priscilawebdev** [comment]
I'm closing this until we reach a decision on possible changes to the issue's page

**Pull request closed without merging**